modules/tfc-oidc/variables.tf

```
@@ -0,0 +1,127 @@
+  default     = "Workload Identity Pool managed by Terraform"
+}
+
+variable "provider_id" {
+  type        = string
+  description = "Workload Identity Pool Provider id"
+}
+
+variable "provider_display_name" {
+  type        = string
+  description = "Workload Identity Pool Provider display name"
+  default     = null
+}
+
+variable "provider_description" {
+  type        = string
+  description = "Workload Identity Pool Provider description"
+  default     = "Workload Identity Pool Provider managed by Terraform"
+}
+
+variable "attribute_condition" {
+  type        = string
+  description = <<-EOF
+    Workload Identity Pool Provider attribute condition expression. 
+    [More info](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/iam_workload_identity_pool_provider#attribute_condition)
+  EOF
+  default     = ""
+}
+
+variable "attribute_mapping" {
+  type        = map(any)
+  description = <<-EOF
+    Workload Identity Pool Provider attribute mapping. 
+    [More info](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/iam_workload_identity_pool_provider#attribute_mapping)
+    EOF
+  default = {
+    "google.subject"                        = "assertion.sub",
+    "attribute.aud"                         = "assertion.aud",
+    "attribute.terraform_run_phase"         = "assertion.terraform_run_phase",
+    "attribute.terraform_project_id"        = "assertion.terraform_project_id",
+    "attribute.terraform_project_name"      = "assertion.terraform_project_name",
+    "attribute.terraform_workspace_id"      = "assertion.terraform_workspace_id",
+    "attribute.terraform_workspace_name"    = "assertion.terraform_workspace_name",
+    "attribute.terraform_organization_id"   = "assertion.terraform_organization_id",
+    "attribute.terraform_organization_name" = "assertion.terraform_organization_name",
+    "attribute.terraform_run_id"            = "assertion.terraform_run_id",
+    "attribute.terraform_full_workspace"    = "assertion.terraform_full_workspace",
+  }
+}
+
+variable "allowed_audiences" {
+  type        = list(string)
+  description = "Workload Identity Pool Provider allowed audiences."
+  default     = []
+}
+
+variable "issuer_uri" {
+  type        = string
+  description = <<-EOF
+    Workload Identity Pool Issuer URL for Terraform Cloud/Enterprise. The default audience format used by TFC is of the form
+    //iam.googleapis.com/projects/{project number}/locations/global/workloadIdentityPools/{pool ID}/providers/{provider ID}
+    which matches with the default accepted audience format on GCP.
+  EOF
+  default     = "https://app.terraform.io"
+}
+
+variable "tfc_organization_name" {
+  type        = string
+  description = "The name of your Terraform Cloud organization"
+}
+
+variable "tfc_project_name" {
+  type        = string
+  default     = "Default Project"
+  description = "The project under which a workspace will be created"
```

Module isn't creating a workspace.

```suggestion
  description = "The Terraform Cloud project to authorize via OIDC"
```